src/components/product/Product.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import './product.scss';
+
+class Product extends Component {
+
+    constructor(props) {
+
+        super(props);
+
+        this.fields = [
+            { label: 'title', reg: /^.{3,}$/ },
+            { label: 'price', reg: /^[0-9]+$/ },
+            { label: 'description', reg: /^.{3,}$/ },
+        ];
+
+        this.state = {};
+
+        this.fields.forEach(({ label }) => this.state[label] = ({
+            value: '',
+            error: false,
+            display: false,
+        }));
+    }
+
+    validate = ({ target }) => {
+        const field = this.fields.find(item => item.label === target.name);
+        const stateField = this.state[target.name];
+
+        !field.reg.test(target.value) ?
+            this.setState({
+                [target.name]: {
+                    ...stateField,
+                    error: target.name === 'price' ? "Should be only digits" : "Is not valid"
+                }
+            }) :
+            this.setState({ [target.name]: { ...stateField, error: false, display: true } });
+    }
+
+    handleInputChange = e => {
+        const field = this.state[e.target.name];
+        this.setState({ [e.target.name]: { ...field, value: e.target.value } });
+    }
+
+    handleSpanBlur = e => {
+        const field = this.state[e.target.id];
+        this.setState({ [e.target.id]: { ...field, display: false } });
+    }
+
+    getButtonState() {
+        return Object.entries(this.state).some(([key, { error, value }]) => error || !value);
+    }
+
+    onSubmit = (e) => {
+        e.preventDefault();
+        const data = {};
+        Object.entries(this.state).forEach(([key, { value }]) => data[key] = value);
+
+    };
+
+    // TODO: need autofocus
+
+    render() {
+
+        const { title, price, description } = this.state;
+
+        const htmlTitle = !title.display ?
+
+            <h3>TITLE: <input name="title"
```

Comment:
actually, here we need custom component span/input, but not a simple input

Reply:
In this case we have a question with validation. Where it should be: in the current component or child span/input ? What is the best way ?

Reply:
It is up to you. 
Your custom input can take additional optional prop and make validation according to it (`<CustomInput onBlur={...} validation={/.../}`).
Or you can make validation onBlur directly in parent component. I would stick with this approach.

Reply:
My proposition is leave validation in Product component and push this line through the props to child component
https://github.com/ex3emxl/AJS-tasks/blob/shop/src/components/product/Product.js#L89
Is it good idea, what do you think ?

Reply:
yes, it is good enough